extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,21 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    const selector = config['selector'] || (spec && spec['selector']);
+    const analyticsVariables = config[vars] || {};
+    if (spec) {
       if (!isVisibilitySpecValid(config)) {
         return;
       }
+      const el = this.win_.document.getElementById(selector
```

Comment:
What I mean is that you can do this just before Line 231. The difference is that in one case, the variables are extracted during config registeration while in the other case, the variables are extracted when the trigger fires.